pages/mesosphere/dcos/services/edge-lb/1.5/concepts/edge-lb-architecture-deep-dive/index.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: layout.pug
+navigationTitle: Architectural deep-dive
+title: Architectural deep-dive
+menuWeight: 20
+excerpt: Provides a more detailed view of Edge-LB components within the network topology for a cluster
+enterprise: true
+---
+
+Edge-LB provides layer-7 inbound load balancing and proxying through the Edge-LB API server and Edge-LB pools. This section provides a deeper view into the details of the Edge-LB architecture and how it works.
+
+The following diagram provides a more detailed representation of the Edge-LB architecture.
+
+![Detail view of Edge-LB components](/mesosphere/dcos/services/edge-lb/1.5/img/Edge-LB-detail-arch.png)
+
+# A closer look at the Edge-LB API server
+The Edge-LB API receives incoming client requests and, based on the action requested, creates, deletes, or updates Edge-LB pool instances. The API server also serves API endpoint requests for Edge-LB pool(s) instances.
+
+The Edge-LB API server uses the DC/OS ZooKeeper persistent storage to store the configuration file for the load balancer and communicates to the Edge-LB pool instances through ZooKeeper&reg; transactions. The interaction between ZooKeeper and the Edge-LB API server enables Edge-LB to be fault tolerant. In the event of an API server failure, the Edge-LB pool instances continue to load-balance traffic to the proper backend. When the Edge-LB API server comes back online, it re-establishes the communication between itself and the pool instances.
+
+The Edge-LB API server has two sub-components:
+- Apache&reg; Mesos&reg; listener
+- DC/OS template
+
+## Mesos listener
+The `mesos-listener` sub-component is always up by default. It subscribes to the Mesos event stream, translates the event stream data into an internal data structure, and serves metadata from the data structure through the API server. It also runs a gRPC server to serve the Mesos task information to other Edge-LB components like the `dcos-template`.
+
+## DC/OS template
+The `dcos-template` sub-component dynamically renders the load balancer configuration information for HAProxy&reg; by combining the `mesos-listener` output and the Edge-LB pool configuration provided by the Edge-LB CLI client. The template then signals the `lbmgr` sub-component to take appropriate action on the HAProxy load balancer instance.
+
+In the event of Edge-LB pool update, the `dcos-template` re-generates the HAProxy load balancer configuration from the Edge-LB pool configuration file and signals `lbmgr` to reload the existing HAProxy configuration in real-time.
```

Comment:
> and signals `lbmgr` to reload the existing HAProxy configuration in real-time.

It is not dcos-template but cloud-controller, and there is a persistent, streaming gRPC connection between cloud-controller and `lbmgr` through which the new configuration is sent. We used to do signaling but IIRC in 1.5, we switched to gRPC.

Also, the `lbmgr` in this context may confuse the reader, as it is a pretty deep implementation detail. `lbmgr` is a part of the pool server, it supervises the proxy process and coordinates reloads. Having said that, what do you think about using the term `pool server` here instead and describing the role of lbmgr later on in the text? 

I am not sure how well the term `pool server` is established in the documentation. Even within the team we have troubles using consistent naming for it. Maybe it would be nice to fix it now and agree upon`pool server`?

Reply:
Created separate ticket